pkg/tinydtls/contrib/sock_dtls.c

```diff
@@ -452,4 +492,46 @@ static void _session_to_ep(const session_t *session, sock_udp_ep_t *ep)
     memcpy(&ep->addr.ipv6, &session->addr, sizeof(ipv6_addr_t));
 }
 
+#ifdef SOCK_HAS_ASYNC
+void _udp_cb(sock_udp_t *sock, sock_async_flags_t flags)
+{
+    if (flags & SOCK_ASYNC_MSG_RECV) {
+        sock_dtls_session_t remote;
+        ssize_t res = sock_udp_recv(sock, data, max_len, timeout,
+                                    &remote.ep);
```

Comment:
Another issue: Where is the buffer here coming from. We could provide a buffer on the stack, but I think this is undesirable. Likewise, is a global buffer for something that is basically just a wrapper layer (and would require 3 buffer spaces: the network stack's, sock DTLS's, and the application's buffer which is ... undesirable). I think I am going to propose a `sock` amendment that allows for zero-copy, that @kaspar030, @haukepetersen, and I discussed some while back, that would solve this situation here as well.

Reply:
See https://github.com/RIOT-OS/RIOT/pull/13616.